features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/internal/tracking/AndroidXFragmentLifecycleCallbacks.kt

```diff
@@ -77,11 +87,19 @@ internal open class AndroidXFragmentLifecycleCallbacks(
         }
     }
 
-    override fun onFragmentPaused(fm: FragmentManager, f: Fragment) {
-        super.onFragmentPaused(fm, f)
-        componentPredicate.runIfValid(f, internalLogger) {
-            val key = resolveKey(it)
-            rumMonitor.stopView(key)
+    @MainThread
+    override fun onFragmentStopped(fm: FragmentManager, f: Fragment) {
+        super.onFragmentStopped(fm, f)
+        executor.scheduleSafe(
+            "Delayed view stop",
+            STOP_VIEW_DELAY_MS,
+            TimeUnit.MILLISECONDS,
+            sdkCore.internalLogger
+        ) {
+            componentPredicate.runIfValid(f, internalLogger) {
```

Comment:
what happens if we delay the new view resumes/starts in before the 150ms interval ? In that case this scheduled action will not just stop it ?

Reply:
I'm not sure I understand the question you're asking. Why would we delay the new view resume/start?

Reply:
let me rephrase this, if I am correct here you are postponing the `stopView` method call for the current view with 150ms. In case the next view `onResume` is called during this interval, your `stopView` will be called just after. I wanted to make sure in that case you are not going to stop the `just started new view`. The way I remember we are doing a validation on the event.key but I am not sure anymore that's the reason wanted to make sure you verified that.

Reply:
The stop view can only stop a view created with the same key, meaning we can't stop the newly created view.